app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,202 @@
+      };
+      const payload = {
+        recipeName: recipeFormData.recipeName,
+        description: recipeFormData.description,
+        directives: recipeFormData.directives,
+      };
+      MyDataPrepApi.updateRecipe(params, payload).subscribe(
+        () => {
+          setIsNameError(false);
+          setIsCreateAndEditRecipeFormOpen(false);
+          setSnackbar({
+            open: true,
+            isSuccess: true,
+            message: `${recipeSteps.length} ${T.translate(
+              'features.WranglerNewUI.RecipeForm.labels.recipeSaveSuccessMessage'
+            )}`,
+          });
+        },
+        (err) => {
+          if (err.response.message) {
+            setIsNameError(true);
+          } else {
+            setIsCreateAndEditRecipeFormOpen(true);
+            setSnackbar({
+              open: true,
+              isSuccess: false,
+              message: T.translate(
+                'features.WranglerNewUI.RecipeForm.labels.errorMessage'
+              ).toString(),
+            });
+          }
+        }
+      );
+    }
+  };
+
+  return (
+    <StyledForm
+      onSubmit={(event: FormEvent) => onFormHandle(event)}
+      data-testid="recipe-form-parent"
+    >
+      <FormFieldWrapper>
+        <StyledLabel data-testid="recipe-name-label">
+          {T.translate('features.WranglerNewUI.RecipeForm.labels.name')}
+        </StyledLabel>
+        <StyledRecipeNameTextField
+          required
+          variant="outlined"
+          defaultValue={recipeData.recipeName}
+          error={isNameError}
+          id="outlined-error-helper-text"
+          helperText={
+            isNameError
+              ? T.translate('features.WranglerNewUI.RecipeForm.labels.nameErrorMessage')
+              : ''
+          }
+          fullWidth
+          onChange={(event) =>
+            setRecipeFormData({ ...recipeFormData, ['recipeName']: event.target.value })
+          }
+          data-testid="recipe-name-field"
+          placeholder={T.translate('features.WranglerNewUI.RecipeForm.labels.namePlaceholder')}
+        />
+      </FormFieldWrapper>
+      <FormFieldWrapper>
+        <FormControl variant="outlined">
+          <NormalLabel data-testid="recipe-description-label">
+            {T.translate('features.WranglerNewUI.RecipeForm.labels.description')}
+          </NormalLabel>
+          <StyledDescriptionTextArea
+            required
+            aria-label="minimum height"
+            minRows={3}
+            data-testid="recipe-description-field"
+            defaultValue={recipeData.description}
+            onChange={(event) =>
+              setRecipeFormData({ ...recipeFormData, ['description']: event.target.value })
+            }
+            placeholder={T.translate(
+              'features.WranglerNewUI.RecipeForm.labels.descriptionPlaceholder'
+            )}
+          />
+        </FormControl>
+      </FormFieldWrapper>
+      <StyledFormButtonWrapper>
+        <StyledCancelButton
```

Comment:
There are Different types of Button components available in cdap-ui under shared components, Can't we use them here?

Reply:
now we've used shared components